aiida_vasp/utils/delegates.py

```diff
@@ -0,0 +1,56 @@
+"""Module containing various decorators implementing delegate types."""
+from functools import update_wrapper
+
+
+def delegate_method_kwargs(prefix='_init_with_'):
```

Comment:
This also needs a test (aiida_vasp/utils/tests/test_delegates.py). It should also be a usage example.

Reply:
Added a test for the `@delegate()` decorator and extended the corresponding doctoring with an example usage.